service/lib/dinstaller/dbus/interfaces/dasd.rb

```diff
@@ -124,7 +123,17 @@ def dasd_set_diag(paths, diag)
         # @param paths [Array<String>]
         # @return [Integer]
         def dasd_format(paths)
-          dasds_dbus_method(paths) { |dasds| dasd_backend.format(dasds) }
+          dasds = find_dasds(paths)
+          return [1, ""] if dasds.nil?
+
+          job = nil
+          progress = proc { |statuses| job.update_format(statuses) }
```

Comment:
How does this work, are we calling `update_format` on `nil`?

Reply:
Not really. The callback is only called by the child thread after that thread has slept, so the initialization below has happened "for sure".

Reply:
Clarified with comments.